packages/@ourworldindata/grapher/src/stackedCharts/AbstractStackedChart.tsx

```diff
@@ -266,25 +265,22 @@ export class AbstractStackedChart
         return ""
     }
 
-    @computed private get colorScheme(): ScaleOrdinal<string, string> {
-        const scheme =
-            (this.manager.baseColorScheme
-                ? ColorSchemes[this.manager.baseColorScheme]
-                : null) ?? ColorSchemes.stackedAreaDefault
+    @computed private get categoricalColorAssigner(): CategoricalColorAssigner {
         const seriesCount = this.isEntitySeries
             ? this.selectionArray.numSelectedEntities
             : this.yColumns.length
-        const baseColors = scheme.getColors(seriesCount)
-        if (this.manager.invertColorScheme) baseColors.reverse()
-        return scaleOrdinal(baseColors)
-    }
-
-    getColorForSeries(seriesName: SeriesName): string {
-        const table = this.transformedTable
-        const color = this.isEntitySeries
-            ? table.getColorForEntityName(seriesName)
-            : table.getColorForColumnByDisplayName(seriesName)
-        return color ?? this.colorScheme(seriesName) ?? "#ddd"
+        return new CategoricalColorAssigner({
+            colorScheme:
+                (this.manager.baseColorScheme
+                    ? ColorSchemes[this.manager.baseColorScheme]
+                    : null) ?? ColorSchemes.stackedAreaDefault,
```

Comment:
Much easier on the eye, thanks!

Reply:
Ahh, but now Typescript complains because `this.manager.baseColorScheme` might be undefined. Maybe `this.manager.baseColorScheme ? ColorSchemes[this.manager.baseColorScheme] : ColorSchemes.stackedAreaDefault` then?

Reply:
Oh, right.
No, we cannot do that, that's a change in logic that can potentially be harmful - it could happen that `baseColorScheme` is not a key inside `ColorSchemes`, and in that case we also want to fall back to the default scheme.

Reply:
Ah, you're right. So, leave as is?